src/NuGet.Clients/PackageManagement.VisualStudio/VSSettings.cs

```diff
@@ -74,12 +80,12 @@ private void OnSolutionOpenedOrClosed(object sender, EventArgs e)
 
         public bool DeleteSection(string section)
         {
-            return SolutionSettings.DeleteSection(section);
+            return CanChangeSettings && SolutionSettings.DeleteSection(section);
```

Comment:
It is currently impossible for LoadDefaultSetting to return null without an exception.
There's already a big comment above the "return NullSettings.Instance"